File: platform/Apple/Sources/SweetEditorCoreInternal/core/CoreInteraction.swift
import Foundation

public enum EventType: Int32 {
    case UNDEFINED = 0
    case TOUCH_DOWN = 1
    case TOUCH_POINTER_DOWN = 2
    case TOUCH_MOVE = 3
    case TOUCH_POINTER_UP = 4
    case TOUCH_UP = 5
    case TOUCH_CANCEL = 6
    case MOUSE_DOWN = 7
    case MOUSE_MOVE = 8
    case MOUSE_UP = 9
    case MOUSE_WHEEL = 10
    case MOUSE_RIGHT_DOWN = 11
    case DIRECT_SCALE = 12
    case DIRECT_SCROLL = 13

    public static func fromValue(_ value: Int32) -> EventType {
        switch value {
        case 0: return .UNDEFINED
        case 1: return .TOUCH_DOWN
        case 2: return .TOUCH_POINTER_DOWN
        case 3: return .TOUCH_MOVE
        case 4: return .TOUCH_POINTER_UP
        case 5: return .TOUCH_UP
        case 6: return .TOUCH_CANCEL
        case 7: return .MOUSE_DOWN
        case 8: return .MOUSE_MOVE
        case 9: return .MOUSE_UP
        case 10: return .MOUSE_WHEEL
        case 11: return .MOUSE_RIGHT_DOWN
        case 12: return .DIRECT_SCALE
        case 13: return .DIRECT_SCROLL
        default: return .UNDEFINED
        }
    }
}

public enum GestureType: Int32 {
    case UNDEFINED = 0
    case TAP = 1
    case DOUBLE_TAP = 2
    case LONG_PRESS = 3
    case SCALE = 4
    case SCROLL = 5
    case FAST_SCROLL = 6
    case DRAG_SELECT = 7
    case CONTEXT_MENU = 8

    public static func fromValue(_ value: Int32) -> GestureType {
        switch value {
        case 0: return .UNDEFINED
        case 1: return .TAP
        case 2: return .DOUBLE_TAP
        case 3: return .LONG_PRESS
        case 4: return .SCALE
        case 5: return .SCROLL
        case 6: return .FAST_SCROLL
        case 7: return .DRAG_SELECT
        case 8: return .CONTEXT_MENU
        default: return .UNDEFINED
        }
    }
}

public enum HitTargetType: Int32 {
    case NONE = 0
    case INLAY_HINT_TEXT = 1
    case INLAY_HINT_ICON = 2
    case GUTTER_ICON = 3
    case FOLD_PLACEHOLDER = 4
    case FOLD_GUTTER = 5
    case INLAY_HINT_COLOR = 6
    case CODELENS = 7
    case LINK = 8

    public static func fromValue(_ value: Int32) -> HitTargetType {
        switch value {
        case 0: return .NONE
        case 1: return .INLAY_HINT_TEXT
        case 2: return .INLAY_HINT_ICON
        case 3: return .GUTTER_ICON
        case 4: return .FOLD_PLACEHOLDER
        case 5: return .FOLD_GUTTER
        case 6: return .INLAY_HINT_COLOR
        case 7: return .CODELENS
        case 8: return .LINK
        default: return .NONE
        }
    }
}

public struct GestureEvent {
    public var type: EventType = .UNDEFINED
    public var points: [PointF] = []
    public var modifiers: Int32 = KeyModifier.NONE
    public var wheel_delta_x: Float = 0
    public var wheel_delta_y: Float = 0
    public var direct_scale: Float = 1

    public init(type: EventType = .UNDEFINED, points: [PointF] = [], modifiers: Int32 = KeyModifier.NONE, wheel_delta_x: Float = 0, wheel_delta_y: Float = 0, direct_scale: Float = 1) {
        self.type = type
        self.points = points
        self.modifiers = modifiers
        self.wheel_delta_x = wheel_delta_x
        self.wheel_delta_y = wheel_delta_y
        self.direct_scale = direct_scale
    }
}

public struct HitTarget {
    public var type: HitTargetType = .NONE
    public var line: Int32 = 0
    public var column: Int32 = 0
    public var icon_id: Int32 = 0
    public var color_value: Int32 = 0

    public init(type: HitTargetType = .NONE, line: Int32 = 0, column: Int32 = 0, icon_id: Int32 = 0, color_value: Int32 = 0) {
        self.type = type
        self.line = line
        self.column = column
        self.icon_id = icon_id
        self.color_value = color_value
    }
}

File: platform/Flutter/sweeteditor/lib/core/core_interaction.dart
// ignore_for_file: unused_element

part of 'editor_core.dart';

enum EventType {
  undefined(0),
  touchDown(1),
  touchPointerDown(2),
  touchMove(3),
  touchPointerUp(4),
  touchUp(5),
  touchCancel(6),
  mouseDown(7),
  mouseMove(8),
  mouseUp(9),
  mouseWheel(10),
  mouseRightDown(11),
  directScale(12),
  directScroll(13),
  directGestureBegin(14),
  directGestureEnd(15);

  const EventType(this.value);
  final int value;

  static EventType fromValue(int value) {
    switch (value) {
      case 0: return undefined;
      case 1: return touchDown;
      case 2: return touchPointerDown;
      case 3: return touchMove;
      case 4: return touchPointerUp;
      case 5: return touchUp;
      case 6: return touchCancel;
      case 7: return mouseDown;
      case 8: return mouseMove;
      case 9: return mouseUp;
      case 10: return mouseWheel;
      case 11: return mouseRightDown;
      case 12: return directScale;
      case 13: return directScroll;
      case 14: return directGestureBegin;
      case 15: return directGestureEnd;
      default: throw ArgumentError.value(value, 'value', 'Unknown EventType value');
    }
  }
}

enum GestureType {
  undefined(0),
  tap(1),
  doubleTap(2),
  longPress(3),
  scale(4),
  scroll(5),
  fastScroll(6),
  dragSelect(7),
  contextMenu(8);

  const GestureType(this.value);
  final int value;

  static GestureType fromValue(int value) {
    switch (value) {
      case 0: return undefined;
      case 1: return tap;
      case 2: return doubleTap;
      case 3: return longPress;
      case 4: return scale;
      case 5: return scroll;
      case 6: return fastScroll;
      case 7: return dragSelect;
      case 8: return contextMenu;
      default: throw ArgumentError.value(value, 'value', 'Unknown GestureType value');
    }
  }
}

enum HitTargetType {
  none(0),
  inlayHintText(1),
  inlayHintIcon(2),
  inlayHintColor(3),
  codelens(4),
  link(5),
  gutterIcon(6),
  foldGutter(7),
  foldPlaceholder(8);

  const HitTargetType(this.value);
  final int value;

  static HitTargetType fromValue(int value) {
    switch (value) {
      case 0: return none;
      case 1: return inlayHintText;
      case 2: return inlayHintIcon;
      case 3: return inlayHintColor;
      case 4: return codelens;
      case 5: return link;
      case 6: return gutterIcon;
      case 7: return foldGutter;
      case 8: return foldPlaceholder;
      default: throw ArgumentError.value(value, 'value', 'Unknown HitTargetType value');
    }
  }
}

class GestureEvent {
  const GestureEvent({
    this.type = EventType.undefined,
    this.points = const [],
    this.modifiers = 0,
    this.wheelDeltaX = 0.0,
    this.wheelDeltaY = 0.0,
    this.directScale = 1.0,
  });

  final EventType type;
  final List<PointF> points;
  final int modifiers;
  final double wheelDeltaX;
  final double wheelDeltaY;
  final double directScale;
}

class HitTarget {
  const HitTarget({
    this.type = HitTargetType.none,
    this.line = 0,
    this.column = 0,
    this.iconId = 0,
    this.colorValue = 0,
  });

  final HitTargetType type;
  final int line;
  final int column;
  final int iconId;
  final int colorValue;
}
